src/main/java/PersonalBudget/business/balance/domain/service/BalancePageHandler.java

```diff
@@ -0,0 +1,96 @@
+    private static final String MENU_PAGE = "menu/main";
+
+
+    public String handleCurrentMonthBalancePage(Model model) {
+        LocalDate dateFrom = getFirstDayCurrentMonth();
+        LocalDate dateTo= getLastDayCurrentMonth();
+        handleBalancePage(model, dateFrom, dateTo);
+        return BALANCE_PAGE;
+    }
+
+    public String handlePreviousMonthBalancePage(Model model) {
+        LocalDate dateFrom = getFirstDayPreviousMonth();
+        LocalDate dateTo = getLastDayPreviousMonth();
+        handleBalancePage(model, dateFrom, dateTo);
+        return BALANCE_PAGE;
+    }
+
+    public String handleCurrentYearBalancePage(Model model) {
+        LocalDate dateFrom = getFirstDayCurrentYear();
+        LocalDate dateTo = getLastDayCurrentYear();
+        handleBalancePage(model, dateFrom, dateTo);
+        return BALANCE_PAGE;
+    }
+
+    public String handleNonstandardBalancePage(NonstandardDateDTO nonstandardDateDTO, Model model) {
+        LocalDate dateFrom = nonstandardDateDTO.dateFrom();
+        LocalDate dateTo = nonstandardDateDTO.dateTo();
+        if(dateTo.isBefore(dateFrom)) {
+            balanceTemplateService.addTwoDatesComparisonAttribute(model);
+            return MENU_PAGE;
+        }
+        handleBalancePage(model, dateFrom, dateTo);
+        return BALANCE_PAGE;
+    }
+
+    public String handleNonstandardBalancePageWithoutDateRange(Model model) {
+        balanceTemplateService.addNoDateRangeAttribute(model);
+        return MENU_PAGE;
+    }
+
+    public void handleBalancePage(Model model, LocalDate dateFrom, LocalDate dateTo) {
+        List<CategorySumDTO> incomeCategoriesSum  = balanceGateway.fetchUserIncomeCategoriesSums(dateFrom, dateTo);
+        List<CategorySumDTO> expenseCategoriesSum = balanceGateway.fetchUserExpenseCategoriesSums(dateFrom, dateTo);
+        List<List<Object>> incomeChartData = mapToChartData(incomeCategoriesSum);
+        List<List<Object>> expenseChartData = mapToChartData(expenseCategoriesSum);
+        List<ParticularActivityDTO> particularIncomes = balanceGateway.fetchUserParticularsIncomeCategory(dateFrom, dateTo);
+        List<ParticularActivityDTO> particularExpenses = balanceGateway.fetchUserParticularsExpenseCategory(dateFrom, dateTo);
+        balanceTemplateService.addIncomeParticularAttribute(model, particularIncomes);
+        balanceTemplateService.addExpenseParticularAttribute(model, particularExpenses);
+        balanceTemplateService.addIncomeSumAttribute(model, incomeCategoriesSum);
+        balanceTemplateService.addExpenseSumAttribute(model, expenseCategoriesSum);
+        balanceTemplateService.addIncomeCategoriesSumAttribute(model, incomeCategoriesSum);
+        balanceTemplateService.addExpenseCategoriesSumAttribute(model, expenseCategoriesSum);
+        balanceTemplateService.addIncomeChartDataAttribute(model, incomeChartData);
+        balanceTemplateService.addExpenseChartDataAttribute(model, expenseChartData);
```

Comment:
can you wrap the template service logic in one method? in my opinion the chart data should be build in a mapper, this method does too much than just handling the page.